src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,34 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onTodoClear: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todo,
+  onTodoClear: todoCleaner,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
It seems like you're initializing your state with props. As per the checklist, this is considered an anti-pattern in React. The reason is that the constructor (or useState in this case) only runs once when the component is created, and it won't update if the props change. It's better to handle such cases in useEffect or other lifecycle methods.